### PR TITLE
add k8s 1.26 variants

### LINKED ISSUE #2770: Kubernetes 1.26 support
Kubernetes 1.26 was released in Dec 2022: https://kubernetes.io/blog/2022/12/09/kubernetes-v1-26-release/

[CHANGELOG since 1.25.0](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.26.md#changelog-since-v1250)

- [ ]  aws-k8s variant
- [ ]  vmware-k8s variant
- [ ]  metal-k8s
- [ ]  aws-k8s nvidia variant

### PR DESCRIPTION
**Pull request by gthao313**
**Issue number:**

Closes #2770

**Description of changes:**

```
    add k8s 1.26 variants

    This adds a k8s-1.26 package. Existing 1.25 models are moved to be 1.26,
    with symlinks created for 1.25 models to point to the new locations.
    Finally, new variants are defined for aws-k8s-1.26, aws-k8s-1.26-nvidia,
    metal-k8s-1.26, and vmware-k8s-1.26.
```

**Testing done:**

To be done:

- [x] Update package with final EKS-D 1.26 package source
- [x] Test aws-k8s-1.26 x86_64 joining 1.26 cluster and ran full conformance test. 
- [x] Test aws-k8s-1.26 ARM joining 1.26 cluster and deploying pods
- [x] Test aws-k8s-1.26-nvidia x86-64 joining 1.26 cluster and ran full conformance test
- [x] Test aws-k8s-1.26-nvidia ARM joining 1.26 cluster and ran full conformance test
- [x] Test metal-k8s-1.26 joining 1.25 cluster and ran full conformance test
- [x] Test vmware-k8s-1.26 joining 1.25 cluster and ran full conformance test
- [x] Check kubelet logs and system logs in general, compared them with the aws-k8s-1.25 variant kubelet logs.
- [x] Confirm the vSphere CSI driver still works
- [x] Confirm the EBS CSI driver still works.
- [x] Smoke test aws-k8s-1.26-nvidia variant with some GPU workloads

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by gthao313** [comment]
need update after merging https://github.com/bottlerocket-os/bottlerocket/pull/2774

File: .github/workflows/build.yml
Comment (stmcginnis) [inline .github/workflows/build.yml]: You will pick it up on a rebase, but just watch out for any kind of merge conflict with the changes in variants. The `aws-k8s-1.21` entry is now dropped.

File: packages/kubernetes-1.26/Cargo.toml
Comment (stmcginnis) [inline packages/kubernetes-1.26/Cargo.toml]: This can be updated to 2021 now.

File: packages/kubernetes-1.26/kubelet-exec-start-conf
Comment (stmcginnis) [inline packages/kubernetes-1.26/kubelet-exec-start-conf:15]: Does this need to be external now too since cloud providers are no longer in-tree?

Or maybe [dropped completely](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/)?

> (DEPRECATED: will be removed in 1.24 or later, in favor of removing cloud provider code from Kubelet.)

File: variants/aws-k8s-1.26-nvidia/Cargo.toml
Comment (stmcginnis) [inline variants/aws-k8s-1.26-nvidia/Cargo.toml]: This can be raised to 2021 now.

File: variants/aws-k8s-1.26/Cargo.toml
Comment (stmcginnis) [inline variants/aws-k8s-1.26/Cargo.toml]: 2021

File: variants/metal-k8s-1.26/Cargo.toml
Comment (stmcginnis) [inline variants/metal-k8s-1.26/Cargo.toml]: 2021

File: variants/vmware-k8s-1.26/Cargo.toml
Comment (stmcginnis) [inline variants/vmware-k8s-1.26/Cargo.toml]: 2021

**Review by stmcginnis — COMMENTED** [review]
Just some questions and minor things inline. Looks good!

File: packages/kubernetes-1.26/kubelet-exec-start-conf
Comment (gthao313) [reply packages/kubernetes-1.26/kubelet-exec-start-conf:11]: I was thinking maybe we need to keep this empty to not enable a cloud-provider for metal variants where a cloud manager might not exist?

File: packages/kubernetes-1.26/kubelet-exec-start-conf
Comment (gthao313) [inline packages/kubernetes-1.26/kubelet-exec-start-conf:8]: @stmcginnis  Should I change it to `--cloud-provider {{default "external" settings.kubernetes.cloud-provider}} \` since the earlier versions of k8s should still be using in-tree providers would pick that up.

Comment (stmcginnis) [reply packages/kubernetes-1.26/kubelet-exec-start-conf:8]: I'm actually not sure what the value is supposed to be now with Kubernetes 1.26. We might need to ask someone from the EKS-D team.

File: packages/kubernetes-1.26/kubelet-exec-start-conf
Comment (gthao313) [reply packages/kubernetes-1.26/kubelet-exec-start-conf:15]: We can't remove it and enable to `external` since metal variants where a cloud manager might no exist need empty string `""` on.

File: packages/kubernetes-1.26/kubelet-exec-start-conf
Comment (gthao313) [reply packages/kubernetes-1.26/kubelet-exec-start-conf:8]: The right change would be migrate to `--cloud-provider external` and use empty string `""` if it's `metal-*`.

**Comment by gthao313** [comment]
Blocked by EKS team that enable `cluster-provider=external` on ipv6.

**Review by stmcginnis — APPROVED** [review]
This is looking great - thanks @gthao313!

File: packages/kubernetes-1.26/kubernetes-1.26.spec
Comment (etungsten) [inline packages/kubernetes-1.26/kubernetes-1.26.spec]: nit:
```suggestion
Source0: https://distro.eks.amazonaws.com/kubernetes-1-26/releases/3/artifacts/kubernetes/v%{gover}/kubernetes-src.tar.gz
```

**Review by etungsten — APPROVED** [review]

File: packages/kubernetes-1.26/kubernetes-1.26.spec
Comment (jpmcb) [inline packages/kubernetes-1.26/kubernetes-1.26.spec:89]: FYI to others since I was curious:
> echo "'make generated_files' is deprecated.  Please use hack/update-codegen.sh instead."

And looks like it'll be dropped in k8s 1.29
https://github.com/kubernetes/kubernetes/blob/2dae34c4a2768a8d39422be42eae2b2817a41a7c/build/root/Makefile#L311-L317

File: packages/kubernetes-1.26/kubernetes-1.26.spec
Comment (jpmcb) [inline packages/kubernetes-1.26/kubernetes-1.26.spec]: Nit: in keeping with how other k8s packages are built using `v%{gover}`
```
Source0: https://distro.eks.amazonaws.com/kubernetes-1-26/releases/3/artifacts/kubernetes/v%{gover}/kubernetes-src.tar.gz
```

**Review by jpmcb — APPROVED** [review]
Built a aws-k8s-1.25 variant and deployed an ami
```
cargo make -e BUILDSYS_VARIANT=aws-k8s-1.26
...
cargo make -e BUILDSYS_VARIANT=aws-k8s-1.26 ami
```

👍🏼

**Review by stmcginnis — APPROVED** [review]
Looks good, thanks for getting the URL updated.

**Review by etungsten — APPROVED** [review]

**Pull request merged**
